content/scaling-kubernetes-cluster/examples/nginx.conf

```diff
@@ -0,0 +1,65 @@
+# Set the worker processes based on the number of CPU cores
+# Setting to `auto` will calculate it automatically
+worker_processes auto;
+
+# Number of file descriptors used for Nginx
+# The limit for the maximum descriptors on the server is usually set by the OS.
+# If you don't set descriptors, then OS settings will be used which is by default 2000
+worker_rlimit_nofile 100000;
+
+# Only log critical errors
+error_log /var/log/nginx/error.log crit;
```

Comment:
This makes me think that this example is generic to `nginx` and not specific to kubernetes. 

The right way to scale nginx in kubernetes is to use annotations =)

Reply:
https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md

Reply:
I agree that we should use annotations, but they are mostly to change behavior of Ingresses, not the Nginx engine itself (although some of the settings overlap, but not all).

At the same time, here https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress they offer a setting

```
controller.config - nginx ConfigMap entries

```
designed for the Nginx engine settings

Reply:
this should be a last resort. annotations will get them 99% of what they need to do. as soon as they break from annotations, they are entering no-mans-land. would not recommend it.